### PR TITLE
chore: replace internal string formatting with strings2 library

### PR DESCRIPTION
**Pull request by arran4**
Replaced internal, custom string casing manipulations (like `ToKebabCase` and `SanitizeToIdentifier`) with equivalents provided by the `github.com/arran4/strings2` v0.0.6 library. Used configuration options like `strings2.WithNumberMode(strings2.NumberModeMergeWithWord)` to retain perfect backward compatibility without having to alter expected edge cases in the test suite.

---
*PR created automatically by Jules for task [17854703487679074731](https://jules.google.com/task/17854703487679074731) started by @arran4*

### CODE REVIEW

File: parsers/sanitize.go [hunk 1]
```diff
@@ -10,69 +11,47 @@ import (
...
 	}
 
-	res := builder.String()
-	// Ensure it doesn't start with a digit
-	if len(res) > 0 {
-		r, _ := utf8.DecodeRuneInString(res)
-		if unicode.IsDigit(r) {
-			res = "Cmd" + res
-		}
+	words, err := strings2.Parse(builder.String())
```
[the remark this answers is not in the document]
Comment (arran4) [reply parsers/sanitize.go:37]: > high
> 
> We can simplify `SanitizeToIdentifier` by removing the manual rune-by-rune replacement loop and `strings.Builder`. Since `strings2.Parse` already splits strings into words using non-alphanumeric characters as delimiters, we can pass `name` directly to it.\n\nAdditionally, to maintain perfect backward compatibility (as mentioned in the PR description), we should pass `strings2.WithNumberMode(strings2.NumberModeMergeWithWord)` here as well. This ensures that strings containing numbers (e.g., `v1beta1` or `sha256`) are not split into separate words, which would otherwise cause `ToPascalCase` to produce `V1Beta1` instead of the original `V1beta1`.

@jules
